### PR TITLE
feature: code coverage support

### PR DESCRIPTION
**Pull request by Alizter**

### CODE REVIEW

**Comment by rgrinberg** [comment]
There's some relevant failures in CI

File: dune [hunk 1]
```diff
@@ -15,7 +15,10 @@
   lev-fiber-csexp
   (library
    (name lev_fiber_csexp)
-   (libraries fiber csexp stdune dyn lev_fiber))
```
Comment (rgrinberg) [inline dune:18]: We don't need to instrument anything under `(subdir vendor`.

Comment (Alizter) [reply dune:18]: Yeah this was a rough PR so I could see how interacted in the CI. I will tighten it up later.

**Comment by rgrinberg** [comment]
LGTM.

How come I don't see the coverage github action?

**Comment by Alizter** [comment]
Oh It looks like I added it to the workflow in lev by mistake.

**Comment by rgrinberg** [comment]
There's this weird failure:

>   Error: request to https://api.github.com/repos/ocaml/opam-repository/contents/packages%2Focaml-base-compiler failed, reason: connect ETIMEDOUT 192.30.255.116:443

**Pull request merged**